src/test/java/mate/academy/spring/boot/controllerTests/BookControllerTest.java

```diff
@@ -0,0 +1,289 @@
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    @Autowired
+    private ObjectMapper objectMapper;
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/add-books-to-books-table.sql"));
+        }
+    }
+
+    @AfterEach
+    void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/remove-books-from-books-table.sql")
+            );
+        }
+    }
+
+    @Test
+    @WithMockUser
+    @DisplayName("Should return List BookDtoS")
+    public void testGetAllBooks_shouldReturnListOfBooksDto() throws Exception{
+        List<BookDto> expected = new ArrayList<>();
+        BookDto firstBook = getBookDto("Book 1", "Author 1", BigDecimal.valueOf(20.00));
+        expected.add(firstBook);
+        BookDto secondBook = getBookDto("Book 2", "Author 2", BigDecimal.valueOf(33.00));
+        expected.add(secondBook);
+        MvcResult result = mockMvc.perform(get("/books"))
+                .andExpect(status().isOk())
+                .andReturn();
+        List<BookDto> actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                objectMapper.getTypeFactory().constructCollectionType(List.class, BookDto.class));
+        Assertions.assertNotNull(actual);
+        Assertions.assertFalse(actual.isEmpty());
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
```

Comment:
Pay attention, that method reflectionEquals() returns boolean, so it's not enough just call this method, you also need to use assertTrue(), otherwise test will pass even with incorrect data. Example:
`assertTrue(EqualsBuilder.reflectionEquals(expected, actual, "id"))`

Please check it everywhere